lodepng.cpp

```diff
@@ -3256,33 +3256,28 @@ static void getPixelColorRGBA8(unsigned char* r, unsigned char* g,
   if(mode->colortype == LCT_GREY) {
     if(mode->bitdepth == 8) {
       *r = *g = *b = in[i];
-      if(mode->key_defined && *r == mode->key_r) *a = 0;
-      else *a = 255;
+      *a = 255 * !(mode->key_defined && *r == mode->key_r);
```

Comment:
Why use multiplication where it is unnecessary?:
```c
*a = (mode->key_defined && *r == mode->key_r) ?  0 : 255;
```

Reply:
Each conversion from if-else statement to multiplication in this case removes one jump instruction, creating code that has a few less branches and is also slightly smaller

Reply:
I tested this out with godbolt.org, and it looks like the way you write it doesn't matter, the effect is equivalent and it gets compiled to the same thing:

![image](https://user-images.githubusercontent.com/4254201/230929711-9cd3f292-4175-442a-ae75-d0c186d74db2.png)

So the original with if/else looks more readable I'd say